File: arrayAndHashing9/ContainsDuplicate.java

package arrayAndHashing9;

import java.util.HashSet;

import java.util.Set;

//aproach 1
// Time complexity O(n)
// Space Complexity O(n)
class Solution {
    public boolean containsDuplicate(int[] nums) {
    Set<Integer> in = new HashSet<>();
    for(int num :nums){
        if(in.contains(num)){
            return true;
        }
         in.add(num);
    }
        
       return false;
    }
    public static void main(String[] args) {
        int[] nums = {1,2,3,4,1};
        Solution obj = new Solution();
        
         System.out.println(obj.containsDuplicate(nums)); 

    }
}



// aproach2
// Time complexity O(n^2)
// Space Complexity O(1)
class Solution1 {
    public boolean containsDuplicate(int[] nums) {
         for(int i=0;i<nums.length;i++){
             for(int j=i+1;j<nums.length;j++){
                 if(nums[i]==nums[j]){
                     return true;
                 }
             }
         }
         return false;
    }
    public static void main(String[] args) {
        int[] nums = {1,2,3,4,1};
        Solution1 obj = new Solution1();
        
         System.out.println(obj.containsDuplicate(nums)); 

    }
}

File: LEETCODE128/ContainsDuplicate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LEETCODE128
{
    internal class ContainsDuplicate
    {
        // Approach 2
        //Time Complexity O(n)
        // Space complexity O(n)
        public bool ContainsDuplicate1(int[] nums)
        {
           HashSet<int> set = new HashSet<int>();
             for(int i = 0;i<nums.Length; i++)
            {
                if (set.Contains(nums[i]))
                {
                    return true;
                }
                set.Add(nums[i]);
            }
            return false;


        }
        /*
        static void Main(string[] args)
        {
            ContainsDuplicate obj = new ContainsDuplicate();

            int[] nums = { 1, 1, 1, 3, 3, 4, 3, 2, 4, 2 };
            Console.WriteLine(obj.ContainsDuplicate1(nums));


        }
        */
    }


    /*

    // Approach 1
    //Time Complexity O(n^2)
    // Space complexity O(1)
    public bool ContainsDuplicate1(int[] nums)
    {
        for(int i=0; i<nums.Length; i++)
        {
            for(int j=i+1; j < nums.Length; j++)
            {
                if (nums[i] == nums[j])
                {
                    return true;
                }
            }
        }
        return false;


    }

    static void Main(string[] args)
    {
        ContainsDuplicate obj = new ContainsDuplicate();

        int[] nums = { 1, 1, 1, 3, 3, 4, 3, 2, 4, 2 };
        Console.WriteLine(obj.ContainsDuplicate1(nums));


    }
}


    */

}
